R/models.R

```diff
@@ -1,3 +1,25 @@
+#' Retrieve all models available in the Mistral API
+#'
+#' @return A character vector with the models available in the Mistral API
+#'
+#' @examples
+#' models()
+#'
+#' @export
 models <- function() {
+  base_url <- "https://api.mistral.ai"
 
+  req <- request(base_url) |>
+    req_url_path_append("v1", "models") |>
+    authenticate()
+
+  # How to: Do this request the first time it is used (directly or indirectly)
+  # and cache the response for some time.
+  resp <- req_perform(req) |>
```

Comment:
This could perhaps use the `memoise` 📦  but I'd argue let's keep it simple for now

Reply:
Or in `rlang::env_cache()` according to [this blog post](https://blog.r-hub.io/2021/07/30/cache/) so we won't have to add an extra dependency.

Reply:
Great, I did not know about it 👏  @maelle

Reply:
The "risk" though is what if mistral released a new model during that session 🐞

Reply:
models cache could be invalidated after some time or in case a certain type of error happens. But that's defenetly for later indeed.

Reply:
unsure if relevant but sharing anyway: https://httr2.r-lib.org/reference/req_cache.html